MuseumsOfChicago/GeoBlazor.Sample.Museums/Components/Pages/Home.razor

```diff
@@ -0,0 +1,249 @@
+                }
+            }
+        </ol>
+    </ExpandWidget>
+
+</MapView>
+
+@code {
+
+    [Inject]
+    public required Route Route { get; set; }
+
+    [Inject]
+    public required IWebHostEnvironment WebHostEnvironment { get; set; }
+
+    protected override void OnInitialized()
+    {
+        string filePath = Path.Combine(WebHostEnvironment.WebRootPath, "ChicagoMuseums.csv");
+        using StreamReader reader = new StreamReader(filePath);
+        using CsvReader csvReader = new CsvReader(reader, _csvConfiguration);
+        _records = csvReader.GetRecords<MuseumRecord>().ToList();
+    }
+
+    private async Task OnViewRendered()
+    {
+        if (!_isLoaded)
+        {
+            _isLoaded = true;
+            await LoadMuseums();
+        }
+    }
+
+    private async Task LoadMuseums()
+    {
+        await _museumLayer!.Clear();
+        List<Graphic> graphics = [];
+        PictureMarkerSymbol museumSymbol = new(MuseumIconUrl, 30, 30);
+
+        FieldsPopupContent content = new(
+        [
+            new FieldInfo("Discipline", "Discipline"),
+        new FieldInfo( "StreetAddress", "Address"),
+        new FieldInfo("PhoneNumber", "Phone Number"),
+        new FieldInfo("Website", "Website")
+        ]);
+        PopupTemplate popupTemplate = new("{CommonName}", contents: [content]);
+        foreach (MuseumRecord record in FilteredRecords)
+        {
+            Point point = new Point(record.Longitude, record.Latitude);
+            AttributesDictionary attributes = new(new Dictionary<string, object?>
+            {
+                            {nameof(MuseumRecord.Discipline), record.Discipline},
+                            {nameof(MuseumRecord.CommonName), record.CommonName},
+                            {nameof(MuseumRecord.StreetAddress), record.StreetAddress},
+                            {nameof(MuseumRecord.PhoneNumber), record.PhoneNumber},
+                            {nameof(MuseumRecord.Website), record.Website}
+                        });
+
+            Graphic graphic = new Graphic(point, museumSymbol, popupTemplate, attributes);
+            graphics.Add(graphic);
+        }
+
+        await _museumLayer.Add(graphics);
+    }
+
+
+    private async Task OnFilterChanged(ChangeEventArgs arg)
+    {
+        _filterValue = arg.Value!.ToString();
+        await LoadMuseums();
+    }
+
+    private List<MuseumRecord> FilteredRecords =>
+        string.IsNullOrWhiteSpace(_filterValue)
+            ? _records
+            : _records.Where(r => r.Discipline.Equals(_filterValue)).ToList();
+
+
+    private async Task ToggleRouting()
+    {
+        _routing = !_routing;
+        if (!_routing)
+        {
+            await Clear();
+        }
+        await _popupWidget!.SetProperty(nameof(PopupWidget.AutoOpenEnabled), !_routing);
+    }
+
+    private async Task OnClick(ClickEvent clickEvent)
+    {
+        if (!_routing) return;
+        Graphic stop = new(clickEvent.MapPoint);
```

Comment:
Given that so much of the logic is so nicely split out into intuitive little methods, might be nice to pull the routing out into one as well.  Though if that doesn't work for the demo, I totally understand.  When I would do live coding demos, I would often have a bunch of logic sequentially in a method that I would ordinarily refactor into multiple because it can get really trick to jump around to different methods and have it flow well.